Project won't build if JavaScript chosen as Language

## Description

Building the project when JavaScript is selected (instead of the default, TypeScript) when creating the app via `npx create-remix@experimental-netlify-edge` results in a failed build or a compilation error when running `npm run dev`. The reason it fails is because Node.js is unable to resolve a module with a *.ts extension.

Running `npx create-remix@experimental-netlify-edge`. When prompted to select a language, choose JavaScript

```bash
❯ npx create-remix@experimental-netlify-edge       
? Where would you like to create your app? ./my-remix-app
? Do you want me to run `npm install`? Yes
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated

added 988 packages, and audited 989 packages in 5s

198 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
? TypeScript or JavaScript? 
  TypeScript 
❯ JavaScript
```

## Steps to Reproduce

### During Development

Run `npm run dev` or `yarn dev`

```bash
❯ npm run dev
yarn run v1.22.17
$ cross-env NODE_ENV=development netlify dev
◈ Netlify Dev ◈
◈ Ignored general context env var: LANG (defined in process)
◈ No app server detected. Using simple static server
◈ Running static server from "my-remix-app/public"

◈ Static server listening to 3999

   ┌─────────────────────────────────────────────────┐
   │                                                 │
   │   ◈ Server now ready on http://localhost:3000   │
   │                                                 │
   └─────────────────────────────────────────────────┘

Watching Remix app in development mode...

✘ [ERROR] Could not resolve "./server.ts"


Build failed with 1 error:
error: Could not resolve "./server.ts"
💿 Built in 185ms
```

### Deploying via the Netlify CLI

```bash
❯ ntl deploy --build

────────────────────────────────────────────────────────────────
  Netlify Build                                                 
────────────────────────────────────────────────────────────────

❯ Version
  @netlify/build 27.0.1

❯ Flags
  {}

❯ Current directory
  /Users/nicktaylor/dev/my-remix-app

❯ Config file
  /Users/nicktaylor/dev/my-remix-app/netlify.toml

❯ Context
  production

────────────────────────────────────────────────────────────────
  1. build.command from netlify.toml                            
────────────────────────────────────────────────────────────────

$ remix build
Building Remix app in production mode...

✘ [ERROR] Could not resolve "./server.ts"


Build failed with 1 error:
error: Could not resolve "./server.ts"
Error
    at Object.onBuildFailure (/Users/nicktaylor/dev/my-remix-app/node_modules/@remix-run/dev/cli/commands.js:152:13)
    at buildEverything (/Users/nicktaylor/dev/my-remix-app/node_modules/@remix-run/dev/compiler.js:280:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.build (/Users/nicktaylor/dev/my-remix-app/node_modules/@remix-run/dev/compiler.js:105:3)
    at async Object.build (/Users/nicktaylor/dev/my-remix-app/node_modules/@remix-run/dev/cli/commands.js:147:3)
    at async Object.run (/Users/nicktaylor/dev/my-remix-app/node_modules/@remix-run/dev/cli/run.js:462:7)

────────────────────────────────────────────────────────────────
  "build.command" failed                                        
────────────────────────────────────────────────────────────────

  Error message
  Command failed with exit code 1: remix build

  Error location
  In build.command from netlify.toml:
  remix build

  Resolved config
  build:
    command: remix build
    commandOrigin: config
    publish: /Users/nicktaylor/dev/my-remix-app/public
    publishOrigin: config
  headers:
    - for: /build/*
      values:
        Cache-Control: public, max-age=31536000, s-maxage=31536000
  headersOrigin: config
```

--- a/remix.init/index.js
+++ b/remix.init/index.js
@@ -1,0 +1,19 @@
+const fs = require("fs");
+const path = require("path");
+
+module.exports = ({ isTypeScript, rootDirectory }) => {
+  if (isTypeScript === false) {
+    let remixConfigPath = path.join(rootDirectory, "remix.config.js");
+    let remixConfig = require(remixConfigPath);
+    remixConfig.server = "./server.js";
+
+    let newConfig = `
+/**
+ * @type {import('@remix-run/dev').AppConfig}
+ */
+module.exports = ${JSON.stringify(remixConfig, null, 2)}
+`.trim();
+
+    fs.writeFileSync(remixConfigPath, newConfig);
+  }
+};

--- a/remix.init/index.js
+++ b/remix.init/index.js
@@ -2,7 +2,7 @@ const fs = require("fs");
 const path = require("path");
 
 module.exports = ({ isTypeScript, rootDirectory }) => {
-  if (isTypeScript === false) {
+  if (!isTypeScript) {
     let remixConfigPath = path.join(rootDirectory, "remix.config.js");
     let remixConfig = require(remixConfigPath);
     remixConfig.server = "./server.js";

